app/components/Settings/UserConnections/UserConnections.react.js

```diff
@@ -0,0 +1,185 @@
+					// result returned in an array
+					// TODO: add length of paths === 0 check
+					// TODO: add path non null check
+					const path = head(paths);
+					// get the filename to use as username in the logs
+					const splitPath = path.split('/');
+					const fileName = splitPath.length - 1;
+					this.props.updateConnection({
+						[connection]: paths[0],
+						'username': splitPath[fileName]
+					});
+				});
+			}
+		};
+	}
+
+	render() {
+
+		const {connectionObject, updateConnection} = this.props;
+
+        let inputNames = CONNECTION_CONFIG[connectionObject.dialect].map(connection => {
+            return (
+                <div key={connection}>
+                    <span className ={styles.inputName}>{connection}</span>
+                </div>
+            );
+        });
+		let inputs = CONNECTION_CONFIG[connectionObject.dialect]
+			.map(connection => (
+            <div key={connection}>
+                <input className={this.testClass()}
+                    placeholder={this.getPlaceholder(connection)}
+                    type={this.getInputType(connection)}
+                    onChange={e => (
+                        updateConnection({[connection]: e.target.value})
+                    )}
+                    onClick={this.getOnClick(connection)}
+                    value={connectionObject[connection]}
+                    id={`test-input-${connection}`}
+                />
+            </div>
+		));
+
+        let options = null;
+        if (CONNECTION_OPTIONS[connectionObject.dialect].length !== 0) {
+            options = CONNECTION_OPTIONS[connectionObject.dialect]
+                .map((option) => (
+                <div
+                    className={styles.options}
+                    key={option}
+                >
+                    {option}
+                    <label className={styles.label}><input
+                        className={styles.checkbox}
+                        type="checkbox"
+                        onChange={() => {
+                            updateConnection(
+                                {[option]: !configuration.get(option)}
+                            );
+                        }}
+                        id={`test-option-${option}`}
+                    />
+                    </label>
+                </div>
+            ));
+        }
+
+		return (
+            <div>
+
+            <div className={styles.inputContainer}>
+
+                    <div className={styles.inputNamesContainer}>
+                        <span className ={styles.inputName}>{'Documentation'}</span>
+                        {inputNames}
+                    </div>
+                    <div className={styles.inputFieldsContainer}>
+                        <a className={styles.documentationLink}
+                            onClick={() => {
+                            shell.openExternal(documentationLink(connectionObject.dialect));
+                        }}
+                        >
```

Comment:
Should this be the `<Link/>` component? And then can we 🔪  the `require('shell')` stuff above?